ksql-engine/src/main/java/io/confluent/ksql/function/udaf/KudafAggregator.java

```diff
@@ -50,16 +51,35 @@ public GenericRow apply(final Struct k, final GenericRow rowValue, final Generic
     // compute the aggregation and write it into the output row. Its assumed that
     // the columns written by this statement do not overlap with those written by
     // the above statement.
-    aggValToAggFunctionMap.forEach((key, value) ->
+    aggValToAggFunctionMap.forEach((functionIndex, ksqlAggregateFunction) ->
         aggRowValue.getColumns().set(
-            key,
-            value.aggregate(
-                rowValue.getColumns().get(value.getArgIndexInValue()),
-                aggRowValue.getColumns().get(key))));
+            functionIndex,
+            ksqlAggregateFunction.aggregate(
+                rowValue.getColumns().get(ksqlAggregateFunction.getArgIndexInValue()),
+                aggRowValue.getColumns().get(functionIndex))));
 
     return aggRowValue;
   }
 
+  @SuppressWarnings("unchecked")
+  public ValueMapper<GenericRow, GenericRow> getMapper() {
+
+    return (aggRow) -> {
+      final List<Object> columns = Stream.generate(String::new).limit(aggRow.getColumns().size())
```

Comment:
I know you're just taking the code from below, but this is a bizarre way of creating a List! Can you change it and the one below to just create a new ArrayList (filling the values with some sentinel String if necessary)? Also java `Stream` has pretty bad performance characteristics and since this value mapper is applied per record I think it's important to worry about performance and allocations here